common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -4979,10 +4979,11 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
 
     HIVE_QUERY_REEXECUTION_ENABLED("hive.query.reexecution.enabled", true,
         "Enable query reexecutions"),
-    HIVE_QUERY_REEXECUTION_STRATEGIES("hive.query.reexecution.strategies", "overlay,reoptimize",
+    HIVE_QUERY_REEXECUTION_STRATEGIES("hive.query.reexecution.strategies", "overlay,reoptimize,reexecute_lost_am",
         "comma separated list of plugin can be used:\n"
             + "  overlay: hiveconf subtree 'reexec.overlay' is used as an overlay in case of an execution errors out\n"
-            + "  reoptimize: collects operator statistics during execution and recompile the query after a failure"),
+            + "  reoptimize: collects operator statistics during execution and recompile the query after a failure\n"
+            + "  reexecute_lost_am: reexecutes query if it failed due to tez am node gets decommissioned"),
```

Comment:
Not just decommissioned case. We retry, even if AM is killed manually. So, let's keep it "...due to tez AM being killed."

Reply:
No, the plugin does not retry if the AM was killed manually because of the error message that we are grepping for.

In case AM is killed, the error message is something like "*killed by user". It does not throw -100 exitcode.

Reply:
Make sense.